### PR TITLE
Chore: Remove the lodash from being a dependency.

### PR DESCRIPTION
**Pull request by justmiller**

### CODE REVIEW

**Comment by autoric** [comment]
Fixes #28

**Comment by NicholasBoll** [comment]
The CI jobs are failing on peer dependencies. I've updated dependencies and explicitly added peer dependencies as local dependencies in #30

**Comment by justmiller** [comment]
Needs fuz testing for replacing tests

File: .travis.yml [hunk 1]
```diff
@@ -1,8 +1,6 @@
 language: node_js
 node_js:
-  - "0.12"
-  - "0.11"
-  - "0.10"
+  - "6.9"
```
Comment (NicholasBoll) [inline .travis.yml]: I think `"4"` and `"6"` are probably good versions to run tests against. Both are currently under LTS.

**Comment by autoric** [comment]
@justmiller I am seeing a couple of your utility tests failing - can you take a look at that?

File: test/utilitiesTest.coffee [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+  
+  describe 'isArray', ->
+    jsc.property 'same a _ given a value', 'json', assertDeepEqual 'isArray'
+
+    jsc.property 'same for arrays', 'array json', (testValue) ->
+      _.isArray(testValue) == utilities.isArray(testValue)   
+  
+  describe 'isEqual', ->
+    jsc.property 'same for json', 'json', (testValue) ->      
+      _.isArray(testValue) == utilities.isArray(testValue)
```
Comment (NicholasBoll) [inline test/utilitiesTest.coffee]: I think the utility being tested here should be `isEqual` and not `isArray`

File: test/utilitiesTest.coffee [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+
+    jsc.property 'same for arrays', 'array json', (testValue) ->
+      _.isArray(testValue) == utilities.isArray(testValue)
```
Comment (NicholasBoll) [inline test/utilitiesTest.coffee]: Same here

File: src/utilities.coffee [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+      )
+      return acc;
+    ),
+    orig
+  )
+
+isEqual = (left, right) ->
+  if isArray(left)
+    isEqualArray(left, right)
+  else if isString(left) || isFunction(left)
```
Comment (NicholasBoll) [inline src/utilities.coffee]: Are these 2 line necessary? If `isPlainObject` returns `false` for strings and functions (which seems like it should), then the function will work the same without these 2 lines

**Pull request merged**